pkg/virt-controller/watch/vm.go

```diff
@@ -924,6 +935,79 @@ func (c *VMController) stopVMI(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMac
 	return nil
 }
 
+func vmRevisionNamePrefix(vmName string) string {
+	return fmt.Sprintf("revision-start-vm-%s", vmName)
+}
+
+func getVMRevisionName(vmName string, generation int64) string {
+	return fmt.Sprintf("%s-%d", vmRevisionNamePrefix(vmName), generation)
+}
+
+func getVMRevisionData(vm *virtv1.VirtualMachine) (runtime.RawExtension, error) {
```

Comment:
Would probably call
```suggestion
func encodeVMRevisionData(vm *virtv1.VirtualMachine) (runtime.RawExtension, error) {
```
to differentiate between `gets` that make requests/reads and this case where it's just encoding.